### PR TITLE
feat: recipe hook

### PR DESCRIPTION
**Pull request by eoaksnes**
## What does this pull request change?

* Add useRecipe hook

## Why is this pull request needed?

* Get recipe using re-usable hook

### CODE REVIEW

File: packages/dm-core/src/hooks/useRecipe.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+/**
+ * Hook to get a recipe
+ *
+ * @docs Hooks
+ *
+ * @usage
+ * Code example:
+ * ```
+ * import { useRecipe } from '@data-modelling-tool/core'
```
Comment (ingeridhellen) [inline packages/dm-core/src/hooks/useRecipe.tsx:42]:
```suggestion
 * import { useRecipe } from '@development-framework/dm-core'
```

File: packages/dm-core/src/hooks/useRecipe.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+ * }
+ * ```
+ *
+ * @param typeRef - The reference to the blueprint to retrieve a recipe for
+ * @param recipeName - Name of recipe to find (optional)
+ * @returns A list containing the blueprint document, a boolean representing the loading state, and an Error, if any.
+ */
+export const useRecipe = (
+  typeRef: string,
+  recipeName: string | null
```
Comment (ingeridhellen) [inline packages/dm-core/src/hooks/useRecipe.tsx:59]: This way, you would have to send in `recipeName == null ` if you would want to use the default one. Write instead 
```suggestion
  recipeName?: string
```
, then you can skip the input variable if yoou don't want to specify

File: packages/dm-core/src/hooks/useRecipe.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
+import React from 'react'
+import { useBlueprint, TUiRecipe } from '../index'
+
+const findRecipe = (
+  initialUiRecipe: any,
+  recipeName: string | null,
+  recipes: any[]
+) => {
```
Comment (ingeridhellen) [inline packages/dm-core/src/hooks/useRecipe.tsx:8]:
```suggestion
  initialUiRecipe: TUiRecipe | undefined,
  recipeName: string | undefined,
  recipes: TUiRecipe[]
): TUiRecipe => {
```

**Review by soofstad — CHANGES_REQUESTED** [review]

File: packages/dm-core/src/hooks/useRecipe.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    return initialUiRecipe
+  }
+  if (recipes.length > 0) {
+    return recipes[0]
+  }
+  return {
+    name: 'yaml',
+    plugin: 'yaml',
+    config: {},
+  }
```
Comment (ingeridhellen) [inline packages/dm-core/src/hooks/useRecipe.tsx:25]:
```suggestion
  return {
    type: 'CORE:UIRecipe',
    name: 'yaml',
    plugin: 'yaml',
    config: {},
  }
```

**Review by ingeridhellen — CHANGES_REQUESTED** [review]
Looks good! Some minor adjustments on typing and comments, though

File: packages/dm-core/src/hooks/useRecipe.tsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  recipeName: string | null,
+  recipes: any[]
+) => {
+  // If recipe is defined, find and return the ui recipe from available recipes.
+  if (recipeName) {
+    return recipes.find((recipe: any) => recipe.name == recipeName)
+  }
+  // If not recipe is defined, use initialize recipe, or the first from recipes list or lastly fallback.
+  // TODO: how to check that is empty?
+  if (initialUiRecipe) {
```
[the remark this answers is not in the document]
Comment (eoaksnes) [reply packages/dm-core/src/hooks/useRecipe.tsx]: Good catch, I have added a check that the initialUiRecipe keys should have a length > 0.

**Review by ingeridhellen — APPROVED** [review]

**Pull request merged**